bot/cogs/image_dots_cog.py

```diff
@@ -0,0 +1,175 @@
+        # not sure the exact conversion calculation here.
+        asciiHeight = math.ceil( asciiWidth * self.ASCIIXDOTS * ( height / width ) / self.ASCIIYDOTS )
+        # want equal number of asciiDots. i.e. 4x2 array
+        width = asciiWidth * self.ASCIIXDOTS
+        height = asciiHeight * self.ASCIIYDOTS
+        rgb_pixels = rgb_pixels.resize((width,height)).getdata()
+        # conversion of a 1d array to 2d array
+        two_d_array = self.to_matrix(list(rgb_pixels), width)
+        
+        # fix the dimensions to not exceed Discord's character limit
+        while (width * height) / (self.ASCIIXDOTS * self.ASCIIYDOTS) > self.CHAR_LIMIT:
+            width -= self.ASCIIXDOTS
+            height -= self.ASCIIYDOTS
+
+        # now time for the magic
+        finished_image = []
+        
+        # want to make small subsets of the image. i.e. section the image in asciiHeight*asciiWidth
+        # sections.
+        # . .
+        # . .
+        # . .
+        # . .
+        for y in range(0,height, self.ASCIIYDOTS):
+            line_of_braille = ''
+            for x in range(0,width,self.ASCIIXDOTS):
+                # want to get that subsection now. starting in corner (x,y) to (width,height).
+                # in our case we want ASCIIXDOTS width, and ASCIIYDOTS height
+                # referencing: https://stackoverflow.com/questions/38049214/how-to-obtain-a-subarray-in-python-3
+                # we want to get a subset of the image. We can easily use a package like numpy, but
+                # relying on numpy makes the project dependencies huge.
+                line_of_braille += self.image_data_to_braille([sub[x:x+self.ASCIIXDOTS] for sub in two_d_array[y:y+self.ASCIIYDOTS]])
+
+            finished_image.append(line_of_braille)
+        return finished_image
+
+
+
+    @ext.command()
+    @ext.long_help('Takes any image, and returns the brailled image.\nCan specify mobile or pc width characteristics.\nDefault width is mobile size.\
+                    Choose a width of \'pc\', \'mobile\', or leave blank.\n \
+                    Threshold determines which pixels are white, and which are blank. Choose a value [0-255]\n\
+                    The final argument asks, do you want to invert the image or not?')
+    @ext.short_help('Turn an image to a braille image.\nDefault width is mobile size.\nDefault threshold is 150.\n\
+                    To specify threshold you must include all required arguments.\n \
+                    The same goes for all arguments')
+    @ext.example(('todots https://my-cool-image.com/stuff.jpg [mobile|pc] [threshold = 0-255] [inverted = 0/1]',
+    'todots https://my-cool-image.com/stuff.jpg [mobile|pc] [threshold = 0-255]',
+    'todots https://my-cool-image.com/stuff.jpg [mobile|pc]', 'todots https://my-cool-image.com/stuff.jpg'))
+    async def todots(self, ctx, image, device = None, threshold = 150, inverted = 0) -> None:
```

Comment:
inverted should be a boolean converter

Reply:
we should also allow for users to upload images. there is an attachment field on the message object